modules/tinymce/src/plugins/media/main/ts/core/HtmlToData.ts

```diff
@@ -5,89 +5,90 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
-import { Cell, Obj } from '@ephox/katamari';
+import { Obj } from '@ephox/katamari';
 
 import DOMUtils from 'tinymce/core/api/dom/DOMUtils';
-import SaxParser from 'tinymce/core/api/html/SaxParser';
+import DomParser from 'tinymce/core/api/html/DomParser';
+import AstNode from 'tinymce/core/api/html/Node';
+import Schema from 'tinymce/core/api/html/Schema';
 import Tools from 'tinymce/core/api/util/Tools';
 
 import { MediaData } from './Types';
 import { getVideoScriptMatch, VideoScript } from './VideoScript';
 
-type AttrList = Array<{ name: string; value: string }> & { map: Record<string, string> };
-
 const DOM = DOMUtils.DOM;
 
 const trimPx = (value: string): string =>
   value.replace(/px$/, '');
 
-const getEphoxEmbedData = (attrs: AttrList): MediaData => {
-  const style = attrs.map.style;
+const getEphoxEmbedData = (node: AstNode): MediaData => {
+  const style = node.attr('style');
   const styles = style ? DOM.parseStyle(style) : { };
   return {
     type: 'ephox-embed-iri',
-    source: attrs.map['data-ephox-embed-iri'],
+    source: node.attr('data-ephox-embed-iri'),
     altsource: '',
     poster: '',
     width: Obj.get(styles, 'max-width').map(trimPx).getOr(''),
     height: Obj.get(styles, 'max-height').map(trimPx).getOr('')
   };
 };
 
-const htmlToData = (prefixes: VideoScript[], html: string): MediaData => {
-  const isEphoxEmbed = Cell<boolean>(false);
+const htmlToData = (prefixes: VideoScript[], html: string, schema?: Schema): MediaData => {
   let data: any = {};
 
-  SaxParser({
-    validate: false,
-    allow_conditional_comments: true,
-    start: (name, attrs) => {
-      if (isEphoxEmbed.get()) {
-        // Ignore any child elements if handling an EME embed
-      } else if (Obj.has(attrs.map, 'data-ephox-embed-iri')) {
-        isEphoxEmbed.set(true);
-        data = getEphoxEmbedData(attrs);
+  const parser = DomParser({ validate: false, forced_root_block: false }, schema);
+  const rootNode = parser.parse(html);
+
+  for (let node = rootNode; node; node = node.walk()) {
+    if (node.type === 1) {
+      const name = node.name;
+
+      if (node.attr('data-ephox-embed-iri')) {
+        data = getEphoxEmbedData(node);
+        // Don't continue to collect if we find an EME embed
+        break;
       } else {
         if (!data.source && name === 'param') {
-          data.source = attrs.map.movie;
+          data.source = node.attr('movie');
         }
 
         if (name === 'iframe' || name === 'object' || name === 'embed' || name === 'video' || name === 'audio') {
           if (!data.type) {
             data.type = name;
           }
 
-          data = Tools.extend(attrs.map, data);
+          data = Tools.extend(node.attributes.map, data);
```

Comment:
I wonder if we should add a function to return the map instead of using that map object obviously not part of this PR. I have abused that in a few places either for performance or just being lazy. It's also very strange that the map property is stuck into the array so essentially replacing the map function at the time arrays didn't have a map. It should have been it's own _map or something instead on the actual node.

Reply:
Looking at the notes for 7.0, one idea is to remove the AST node implementation entirely which would negate having to worry about any of that. That might not happen, but it's been listed as an idea 🤷

Reply:
That would require a new form of api since raw dom elements doesn't have a way to get the attributes out as a pure object map. Also we would have to add unwrap, wrap etc some where else. I guess that will have to be a 8.0 task seems like a lot of work to go though all that now. 😞